src/App.tsx

```diff
@@ -1,61 +1,132 @@
+  };
+
+  const checkError = () => {
+    if (!newTitle) {
+      setErrorTitle(true);
+    }
+
+    if (!selectedUserName) {
+      setErrorUser(true);
+    }
+  };
+
+  const createTodo = () => {
+    const user = findUser(selectedUserName);
+
+    if (user && newTitle) {
+      const newTodo = {
+        id: maxId,
+        title: newTitle,
+        completed: false,
+        userId: user.id,
+      };
+
+      setCopyTodos(todos => [...todos, newTodo]);
+      setNewTitle('');
+      setSelectedUserName('');
+    }
+  };
+
+  const addHandler
+  = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault();
+    checkError();
+    createTodo();
+  };
+
+  const todos: Todo[] = copyTodos.map(todo => {
+    return {
+      ...todo,
+      user: usersFromServer.find(user => todo.userId === user.id),
+    };
+  });
+
+  useEffect(() => {
+    if (newTitle) {
+      setErrorTitle(false);
+    }
+
+    if (selectedUserName) {
+      setErrorUser(false);
+    }
+  }, [newTitle, selectedUserName]);
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/users" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            Title:
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={newTitle}
+              onChange={(e) => {
+                setNewTitle(e.currentTarget.value);
+              }}
+            />
+          </label>
+          {(errorTitle
+             && <span className="error">Please enter a title</span>)}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
-
-          <span className="error">Please choose a user</span>
+          <label>
+            User:
+            <select
+              data-cy="userSelect"
+              onChange={(e) => setSelectedUserName(e.target.value)}
+              value={!selectedUserName ? '0' : selectedUserName}
+            >
+              {!selectedUserName
+               && (<option value="0" disabled selected>Choose a user</option>)}
+              {usersFromServer.map(user => (
+                <option
+                  key={user.name}
+                  value={user.name}
+                >
+                  {user.name}
+                </option>
+              ))}
+            </select>
+          </label>
+          {errorUser
+           && (<span className="error">Please choose a user</span>)}
         </div>
 
-        <button type="submit" data-cy="submitButton">
+        <button
+          type="submit"
+          data-cy="submitButton"
+          onClick={(e) => addHandler(e)}
```

Comment:
```suggestion
          onClick={(e) => handleAddTodo(e)}
```
Usually, we use the prefix `handle` for handler functions